sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp

```diff
@@ -179,5 +166,26 @@ namespace Azure { namespace Messaging { namespace EventHubs {
       batchEnvelope.MessageFormat = BatchedMessageFormat;
       return batchEnvelope;
     }
+
+    /** @brief Event Data Batch constructor
+     *
+     * @param options Options settings for creating the data batch
+     */
+    EventDataBatch(EventDataBatchOptions options = {})
+        : m_partitionId{options.PartitionId}, m_partitionKey{options.PartitionKey},
+          m_maxBytes{options.MaxBytes}, m_marshalledMessages{}, m_batchEnvelope{}, m_currentSize{0}
+    {
+      if (!options.PartitionId.empty() && !options.PartitionKey.empty())
+      {
+        throw std::runtime_error("Either PartionID or PartitionKey can be set, but not both.");
+      }
+
+      if (options.PartitionId.empty())
+      {
+        m_partitionId = anyPartitionId;
+      }
+    };
+
+    friend _detail::EventDataBatchFactory;
```

Comment:
For some reason, doc generation doesn't like this:
> 2023-08-11T23:02:21.3438487Z D:/a/_work/1/s/sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp(189): error : no matching class member found for [D:\a\_work\1\s\build\sdk\eventhubs\azure-messaging-eventhubs\azure-messaging-eventhubs-docs.vcxproj]
2023-08-11T23:02:21.3439423Z     EventDataBatchFactory
2023-08-11T23:02:21.3439818Z   Possible candidates:
2023-08-11T23:02:21.3440637Z     'Azure::Messaging::EventHubs::_detail::EventDataBatchFactory::EventDataBatchFactory()=delete' at line 69 of file D:/a/_work/1/s/sdk/eventhubs/azure-messaging-eventhubs/src/private/eventhubs_utilities.hpp